cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -325,9 +365,13 @@ func generateMountsModeRun() error {
 		return err
 	}
 	if !isDataMounted {
-		name := filepath.Base(boot.InitramfsUbuntuDataDir)
+		if isDeviceEncrypted("ubuntu-data") {
+			if err := measureRunModeSystem(); err != nil {
```

Comment:
The measurements need to happen on all boots if a TPM is detected, regardless of whether the system being booted has an encrypted volume.

Reply:
@chrisccoulson should the same apply to the locking of keys ?

Reply:
Yes, the same applies to locking of keys